### PR TITLE
Refactor data-services-tests

### PR DESCRIPTION
**Pull request by manoramahp**

### CODE REVIEW

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/ResourcesServiceTestWithSameContextNameTestCase.java [hunk 1]
```diff
@@ -25,14 +25,10 @@
     public void serviceDeployment() throws Exception {
         super.init();
         serviceEndPoint = getServiceUrlHttp(serviceName);
-        deployService(serviceName,
-                new DataHandler(new URL("file:///" +getResourceLocation() + File.separator + "samples"
-                        + File.separator + "dbs" + File.separator
-                        + "rdbms" + File.separator + "ResourcesServiceTestWithSameContextName.dbs")));
+        Assert.assertTrue(isServiceDeployed(serviceName));
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/ResourcesServiceTestWithSameContextNameTestCase.java]: Adding assert message?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/rdf/RDFExposedAsRDFSampleTestCase.java [hunk 2]
```diff
@@ -45,9 +45,7 @@
     public void serviceDeployment() throws Exception {
         super.init();
         String resourceFileLocation = getResourceLocation();
-        deployService(serviceName,
-                new DataHandler(new URL("file:///" + resourceFileLocation +File.separator + "dbs" + File.separator +
-                        "rdf" + File.separator + "RDFExposeAsRDFSample.dbs")));
+        Assert.assertTrue(isServiceDeployed("RDFExposeAsRDFSample"));
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/rdf/RDFExposedAsRDFSampleTestCase.java]: Adding assert message?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/services/NestedQueryTestCase.java [hunk 3]
```diff
@@ -85,6 +91,16 @@ public void listOfficeNestedQueryOperation() throws AxisFault, XPathExpressionEx
         log.info("List Office Nested Query verified");
     }
 
+    @Test(groups = {"wso2.dss"}, description = "nested query empty result test")
+    public void testNestedQueryEmptyResultJson() throws Exception {
+        String endpoint = getServiceUrlHttp(serviceName) + "/listOffices";
+        Map<String, String> headers = new HashMap<>();
+        headers.put("Accept", "application/json");
+        Object[] response = sendGET(endpoint, headers);
+        Assert.assertEquals(response[0], ODataTestUtils.OK);
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/services/NestedQueryTestCase.java]: Adding assert message?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1103DuplicateDataServiceTestCase.java [hunk 4]
```diff
@@ -32,25 +33,25 @@
...
-		sqlFileLis.add(selectSqlFile("CreateTables.sql"));
-		sqlFileLis.add(selectSqlFile("Students.sql"));
-		deployService(serviceName,
-		              createArtifact(getResourceLocation() + File.separator + "dbs" + File.separator + "rdbms" +
-		                             File.separator + "h2" + File.separator + serviceName + ".dbs", sqlFileLis));
+		sqlFileList = new ArrayList<>();
+		sqlFileList.add(selectSqlFile("CreateTables.sql"));
+		sqlFileList.add(selectSqlFile("Students.sql"));
+		SqlDataSourceUtil dataSource = new SqlDataSourceUtil(sessionCookie
+				, dssContext.getContextUrls().getBackEndUrl());
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1103DuplicateDataServiceTestCase.java]: Move the , to just after the sessionCookie in the previous line?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java [hunk 5]
```diff
@@ -58,13 +59,12 @@
...
-        deployService(serviceName,
-                createArtifact(getResourceLocation() + File.separator + "dbs"
-                        + File.separator + "rdbms" + File.separator + "h2"
-                        + File.separator + serviceName + ".dbs", sqlFileLis));
+        List<File> sqlFileList = new ArrayList<File>();
+        sqlFileList.add(selectSqlFile("BoxcarringParallelAccess.sql"));
+        SqlDataSourceUtil dataSource = new SqlDataSourceUtil(sessionCookie
+                , dssContext.getContextUrls().getBackEndUrl());
+        dataSource.createDataSource(sqlFileList);
+        Assert.assertTrue(isServiceDeployed(serviceName));
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java]: Adding assert message?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/CARBON15339JsonGsonFormatterTenantModeTest.java [hunk 6]
```diff
@@ -1,120 +0,0 @@
...
-import java.net.URL;
-import java.util.ArrayList;
-import java.util.List;
-
-/**
- * This test is written to verify the fix for https://wso2.org/jira/browse/CARBON-15339
- * need carbon 4.4.2
- */
-public class CARBON15339JsonGsonFormatterTenantModeTest extends DSSIntegrationTest {
-	private final String serviceName = "H2SimpleJsonTest";
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/CARBON15339JsonGsonFormatterTenantModeTest.java]: Can you please explain why this testcase is removed?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1194NestedQueryEmptyResultTest.java [hunk 7]
```diff
@@ -1,73 +0,0 @@
...
-
-import java.io.File;
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
-import static org.wso2.ei.dataservice.integration.test.odata.ODataTestUtils.sendGET;
-
-public class DS1194NestedQueryEmptyResultTest extends DSSIntegrationTest {
```
Comment (chanikag) [inline integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1194NestedQueryEmptyResultTest.java]: Can you please explain why this testcase is removed?

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/CARBON15339JsonGsonFormatterTenantModeTest.java (hunk 6 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/CARBON15339JsonGsonFormatterTenantModeTest.java]: Moved it here
https://github.com/wso2/product-ei/pull/1311/files#diff-d4682599ea60fba74df1956f44a110bfR116

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1194NestedQueryEmptyResultTest.java (hunk 7 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1194NestedQueryEmptyResultTest.java]: Moved the test case here
https://github.com/wso2/product-ei/pull/1311/files#diff-9413b9cd5ef7fcc560af2025732fe70cR94

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/ResourcesServiceTestWithSameContextNameTestCase.java (hunk 1 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/ResourcesServiceTestWithSameContextNameTestCase.java]: Done

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/services/NestedQueryTestCase.java (hunk 3 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/services/NestedQueryTestCase.java]: Done

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1103DuplicateDataServiceTestCase.java (hunk 4 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1103DuplicateDataServiceTestCase.java]: Done

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java [hunk 8]
```diff
@@ -58,13 +59,12 @@
...
-        sqlFileLis.add(selectSqlFile("BoxcarringParallelAccess.sql"));
-
-        deployService(serviceName,
-                createArtifact(getResourceLocation() + File.separator + "dbs"
-                        + File.separator + "rdbms" + File.separator + "h2"
-                        + File.separator + serviceName + ".dbs", sqlFileLis));
+        List<File> sqlFileList = new ArrayList<File>();
+        sqlFileList.add(selectSqlFile("BoxcarringParallelAccess.sql"));
+        SqlDataSourceUtil dataSource = new SqlDataSourceUtil(sessionCookie
+                , dssContext.getContextUrls().getBackEndUrl());
```
[the remark this answers is not in the document]
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java]: Done

File: integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java (hunk 5 above)
Comment (manoramahp) [reply integration/dataservice-hosting-tests/tests-integration/tests/src/test/java/org/wso2/ei/dataservice/integration/test/jira/issues/DS1083BoxcarringParallelInsertTestCase.java]: Done

**Pull request merged**